### ISSUE TITLE
Unnecessary modify_surface calls

### ISSUE THREAD

**Issue by wmww** [MirServer/mir#1049]
If you run `miral-shell --window-management-trace` and Gedit inside it you'll notice `handle_modify_window` gets called far too often. Somewhere between the Wayland frontend and the window manager it should be detect that our surface spec structs don't actually hold any modifications (or they are resetting values that are identical on the surface). This is a case where generating those structs would be helpful.

**Comment by wmww** [comment]
Fixed by #1056

**Comment by CatalinTermure** [comment]
Hello, I checked out the project recently and noticed that this issue is happening again. Far too many identical calls to `handle_modify_window` are issued for window creation and window movement.

It would be difficult for me to pinpoint the commit or a range where the issue started happening again since I could not build old commits easily. Let me know if you need any more help with reproduction.

Excerpt from opening kitty in a miral-shell session with `--window-management-trace`:
```
[2025-03-29 16:34:58.043880] <information> miral::Window Management: handle_modify_window window_info={name=kitty, type=freestyle, state=restored, top_left=154, 184, size=(989, 947), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0}, modifications={name=bash}
[2025-03-29 16:34:58.043902] <information> miral::Window Management: modify_window window_info={name=kitty, type=freestyle, state=restored, top_left=154, 184, size=(989, 947), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0}, modifications={name=bash}
[2025-03-29 16:34:58.043926] <information> miral::Window Management: ====
[2025-03-29 16:34:58.082632] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.082657] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.082665] <information> miral::Window Management: ====
[2025-03-29 16:34:58.182984] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.183011] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.183020] <information> miral::Window Management: ====
[2025-03-29 16:34:58.283315] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.283342] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.283352] <information> miral::Window Management: ====
[2025-03-29 16:34:58.383639] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.383665] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.383673] <information> miral::Window Management: ====
[2025-03-29 16:34:58.483925] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.483952] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.483961] <information> miral::Window Management: ====
[2025-03-29 16:34:58.584279] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.584304] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.584313] <information> miral::Window Management: ====
[2025-03-29 16:34:58.684535] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.684563] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.684574] <information> miral::Window Management: ====
[2025-03-29 16:34:58.784724] <information> miral::Window Management: handle_modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.784753] <information> miral::Window Management: modify_window window_info={name=, type=freestyle, state=fullscreen, top_left=0, 0, size=(949, 1031), restore_rect=(320, 256, (640, 512)), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0, output_id=1}, modifications={}
[2025-03-29 16:34:58.784762] <information> miral::Window Management: ====
[2025-03-29 16:34:58.795721] <information> miral::Window Management: handle_modify_window window_info={name=bash, type=freestyle, state=restored, top_left=154, 184, size=(989, 947), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0}, modifications={name=chovvy@chovvy-82rf:~/Code/canonical/mir/build/bin/Debug}
[2025-03-29 16:34:58.795759] <information> miral::Window Management: modify_window window_info={name=bash, type=freestyle, state=restored, top_left=154, 184, size=(989, 947), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0}, modifications={name=chovvy@chovvy-82rf:~/Code/canonical/mir/build/bin/Debug}
[2025-03-29 16:34:58.795807] <information> miral::Window Management: ====
[2025-03-29 16:34:58.795879] <information> miral::Window Management: handle_modify_window window_info={name=chovvy@chovvy-82rf:~/Code/canonical/mir/build/bin/Debug, type=freestyle, state=restored, top_left=154, 184, size=(989, 947), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0}, modifications={name=~/Code/canonical/mir/build/bin/Debug}
[2025-03-29 16:34:58.795899] <information> miral::Window Management: modify_window window_info={name=chovvy@chovvy-82rf:~/Code/canonical/mir/build/bin/Debug, type=freestyle, state=restored, top_left=154, 184, size=(989, 947), children={}, exclusive_rect=0, preferred_orientation=0xf, confine_pointer=0}, modifications={name=~/Code/canonical/mir/build/bin/Debug}
```
